binary/reflect.go

```diff
@@ -803,6 +820,24 @@ func writeReflectJSON(rv reflect.Value, rt reflect.Type, w io.Writer, n *int64,
 			WriteTo([]byte("}"), w, n, err)
 		}
 
+	case reflect.Map:
+		WriteTo([]byte("{"), w, n, err)
+		keys := rv.MapKeys()
+		wroteField := false
+		for _, k := range keys {
+			if wroteField {
+				WriteTo([]byte(","), w, n, err)
+			} else {
+				wroteField = true
+			}
+			vRv := rv.MapIndex(k)
+			// json keys are always strings
+			WriteTo([]byte(Fmt("\"%v\":", k.Interface())), w, n, err)
```

Comment:
We should refactor out types/tx.go's jsonEscape and use that to escape k.Interface() as well as opts.JSONName maybe.